src/microsoft-dotnet-interactive/src/htmlKernel.ts

```diff
@@ -87,20 +87,43 @@ export function htmlDomFragmentInserter(htmlFragment: string, configuration?: Ht
     return completionPromise.promise.then(() => container.innerHTML);
 }
 
-export function createHtmlKernelThatWorksWithPageDomInBrowser(config: { kernelName: string, container: HTMLElement, contentBehaviour: "append" | "replace" }): Kernel {
-
-    const kernel = new HtmlKernel(
-        config.kernelName,
-        (htmlFragment: string) => htmlDomFragmentInserter(htmlFragment, {
-            getOrCreateContainer: () => config.container,
-            updateContainerContent: (container, htmlFragment) => {
-                if (config.contentBehaviour === "append") {
-                    container.innerHTML += htmlFragment;
-                } else {
-                    container.innerHTML = htmlFragment;
+export type HtmlKernelInBrowserConfiguration = { kernelName: string, container: HTMLElement | string, contentBehaviour: "append" | "replace" } | { kernelName: string, htmlDomFragmentInserterConfiguration: HtmlDomFragmentInserterConfiguration };
+
+export function createHtmlKernelThatWorksWithPageDomInBrowser(config: HtmlKernelInBrowserConfiguration): Kernel {
```

Comment:
This name is a little verbose if we want to make it public.

Reply:
something like `createHtmlKernelForBrowserHoating`?

Reply:
I think even `createHtmlKernelForBrowser` gets the point across well.

Reply:
thanks!